src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileFormat.cs

```diff
@@ -207,21 +218,48 @@ private LockFile Utf8JsonRead(Stream stream, ILogger log, string path)
         }
 
         [Obsolete]
-        private static LockFile ReadLockFile(JObject cursor, string path)
+        private static LockFile ReadLockFile(JObject cursor, string path, LockFileReadFlags flags)
```

Comment:
This method is marked as `Obsolete`. I think it is better not to modify obsolete methods because when we plan on deleting them in the next .NET major version, it will create more work in the future.

Reply:
This is the fallback for reading the assets file with newtonsoft.json, so until we're ready to remove that code, we need to maintain this codepath as well.

Reply:
Just to make sure I'm parsing these comments correctly, all is good here, right?